### PR TITLE
Feature/a1 mtdc@203 prs api request object validation

### PR DESCRIPTION
**Pull request by cpeeyush**
- Add length validation on prs-api request objects.
- Add test coverage.
- Remove overlapping ITs as now UTs cover such request validations.

### CODE REVIEW

File: coreservices/partsrelationshipservice/integration-tests/src/test/java/net/catenax/prs/integrationtest/GetPartsTreeByVinIntegrationTests.java [hunk 1]
```diff
@@ -52,7 +53,7 @@ public void getPartsTreeByVin_maintainedView_success() {
 
     @Test
     public void getPartsTreeByVin_notExistingVIN_returns404() {
-        var notExistingVin = "not-existing-vin";
+        var notExistingVin = "PQTYHSPVXFGLFLCFM";
```
Comment (algattik) [inline coreservices/partsrelationshipservice/integration-tests/src/test/java/net/catenax/prs/integrationtest/GetPartsTreeByVinIntegrationTests.java:56]: why not faker e.g. faker.regexify()?

File: coreservices/partsrelationshipservice/prs-api/src/test/java/net/catenax/prs/requests/PartsTreeByVinRequestTests.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+                args("Invalid view", b -> b.view(faker.lorem().word()), "view"),
+                args("View not null", b -> b.view(null), "view"),
+                args("View not empty", b -> b.view(EMPTY), "view"),
+                args("View not blank", b -> b.view(faker.regexify(WHITESPACE_REGEX)), "view"),
+
+                args("Aspect not empty", b -> b.aspect(EMPTY), "aspect"),
+                args("Aspect not blank", b -> b.aspect(faker.regexify(WHITESPACE_REGEX)), "aspect"),
+
+                args("Depth min 1", b -> b.depth(faker.number().numberBetween(Integer.MIN_VALUE, 0)), "depth"),
```
Comment (algattik) [inline coreservices/partsrelationshipservice/prs-api/src/test/java/net/catenax/prs/requests/PartsTreeByVinRequestTests.java:44]: is there no max depth here?

File: coreservices/partsrelationshipservice/prs-api/src/test/java/net/catenax/prs/requests/PartsTreeByVinRequestTests.java [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+                args("Vin not null", b -> b.vin(null), "vin"),
+                args("Vin not empty", b -> b.vin(EMPTY), "vin"),
+                args("Vin not blank", b -> b.vin(faker.regexify(WHITESPACE_REGEX)), "vin"),
+                args("Vin must be size 17 [1]", b -> b.vin(faker.lorem().characters(1, 16)), "vin"),
+                args("Vin must be size 17 [2]", b -> b.vin(faker.lorem().characters(18, 100)), "vin")
```
Comment (algattik) [inline coreservices/partsrelationshipservice/prs-api/src/test/java/net/catenax/prs/requests/PartsTreeByVinRequestTests.java:50]: similarly I'd suggest adding explicit tests for 16 and 18 for maximum robustness

File: coreservices/partsrelationshipservice/integration-tests/src/test/java/net/catenax/prs/integrationtest/GetPartsTreeByVinIntegrationTests.java (hunk 1 above)
Comment (cpeeyush) [reply coreservices/partsrelationshipservice/integration-tests/src/test/java/net/catenax/prs/integrationtest/GetPartsTreeByVinIntegrationTests.java]: Correct with faker with fixed length. thanks

File: coreservices/partsrelationshipservice/prs-api/src/test/java/net/catenax/prs/requests/PartsTreeByVinRequestTests.java (hunk 2 above)
Comment (cpeeyush) [reply coreservices/partsrelationshipservice/prs-api/src/test/java/net/catenax/prs/requests/PartsTreeByVinRequestTests.java:44]: No, that is part of service layer logic. So covered by IT.

**Review by algattik — APPROVED** [review]

File: coreservices/partsrelationshipservice/prs-api/src/test/java/net/catenax/prs/requests/PartsTreeByVinRequestTests.java (hunk 3 above)
Comment (cpeeyush) [reply coreservices/partsrelationshipservice/prs-api/src/test/java/net/catenax/prs/requests/PartsTreeByVinRequestTests.java]: Added.

File: coreservices/partsrelationshipservice/prs-api/src/test/java/net/catenax/prs/requests/PartsTreeByVinRequestTests.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (cpeeyush) [reply coreservices/partsrelationshipservice/prs-api/src/test/java/net/catenax/prs/requests/PartsTreeByVinRequestTests.java]: Added.

File: coreservices/partsrelationshipservice/api/prs-v0.1.yaml [hunk 4]
```diff
@@ -17,13 +17,17 @@ paths:
         required: true
         schema:
           type: string
+          maxLength: 10000
```
Comment (marcgs) [inline coreservices/partsrelationshipservice/api/prs-v0.1.yaml:20]: Isn't 10000 a bit excessive?

**Review by marcgs — APPROVED** [review]

File: coreservices/partsrelationshipservice/prs-api/src/test/java/net/catenax/prs/requests/PartsTreeByObjectIdRequestTests.java [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.util.function.UnaryOperator;
+import java.util.stream.Stream;
+
+import static java.util.function.UnaryOperator.identity;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class PartsTreeByObjectIdRequestTests extends RequestTestBase {
```
Comment (marcgs) [inline coreservices/partsrelationshipservice/prs-api/src/test/java/net/catenax/prs/requests/PartsTreeByObjectIdRequestTests.java:13]: Great testing!

File: coreservices/partsrelationshipservice/integration-tests/src/test/java/net/catenax/prs/integrationtest/GetPartsTreeByObjectIdIntegrationTests.java [hunk 6]
```diff
@@ -86,41 +86,6 @@ public void getPartsTreeByObjectId_notExistingOneId_emptyResponse() {
             .body(PART_INFOS, hasSize(0));
     }
 
-    @Test
-    public void getPartsTreeByObjectId_noView_returns400() {
```
Comment (marcgs) [inline coreservices/partsrelationshipservice/integration-tests/src/test/java/net/catenax/prs/integrationtest/GetPartsTreeByObjectIdIntegrationTests.java:90]: IMO we should leave at least 1 test like this to verify that the validation is working E2E

File: coreservices/partsrelationshipservice/api/prs-v0.1.yaml (hunk 4 above)
Comment (algattik) [reply coreservices/partsrelationshipservice/api/prs-v0.1.yaml:20]: This was intentionally done (already in previous validation PRs) to highlight the lack of a requirement in this area. When we define one, we can update this :)

File: coreservices/partsrelationshipservice/integration-tests/src/test/java/net/catenax/prs/integrationtest/GetPartsTreeByObjectIdIntegrationTests.java (hunk 6 above)
Comment (algattik) [reply coreservices/partsrelationshipservice/integration-tests/src/test/java/net/catenax/prs/integrationtest/GetPartsTreeByObjectIdIntegrationTests.java:90]: I think there is one: getPartsTreeByObjectId_exceedMaxDepth_returns400

File: coreservices/partsrelationshipservice/api/prs-v0.1.yaml (hunk 4 above)
Comment (cpeeyush) [reply coreservices/partsrelationshipservice/api/prs-v0.1.yaml:20]: Yes @marcgs I also had the same feeling this 10k is a big number here but we did it this way as we don't have clear requirement here. And more to safe guard api for a very large input but still keeping it big enough in general.

File: coreservices/partsrelationshipservice/prs-api/src/test/java/net/catenax/prs/requests/PartsTreeByObjectIdRequestTests.java (hunk 5 above)
Comment (cpeeyush) [reply coreservices/partsrelationshipservice/prs-api/src/test/java/net/catenax/prs/requests/PartsTreeByObjectIdRequestTests.java:13]: Thank you @marcgs

File: coreservices/partsrelationshipservice/integration-tests/src/test/java/net/catenax/prs/integrationtest/GetPartsTreeByObjectIdIntegrationTests.java (hunk 6 above)
Comment (cpeeyush) [reply coreservices/partsrelationshipservice/integration-tests/src/test/java/net/catenax/prs/integrationtest/GetPartsTreeByObjectIdIntegrationTests.java]: Restored 1 Integration test where request object is being validated before service layer as getPartsTreeByObjectId_exceedMaxDepth_returns400 is done in service layer logic.
